builder/ebssurrogate/builder.go

```diff
@@ -318,6 +318,52 @@ func (b *Builder) Run(ctx context.Context, ui packersdk.Ui, hook packersdk.Hook)
 	amiDevices := b.config.AMIMappings.BuildEC2BlockDeviceMappings()
 	launchDevices := b.config.LaunchMappings.BuildEC2BlockDeviceMappings()
 
+	var imageMethod multistep.Step
+	var volumeMethod multistep.Step
```

Comment:
Given the how far down the use of these steps are I find the name to be too generic. Let me think about this a little but I do wonder if we should either use a `multistep.If(b.config.RootDevice.ImageMethod == "create", StepCreateAMI{...})` or if we should just add both the register and create steps to the steps slice and in each step check the image method to determine if the step should run.

Reply:
Let me know which route you think would be best and I will refactor it.  I attempted to model this on the existing conditional instance step.